### ISSUE TITLE
[BUG] Blazingsql constantly failing and exiting script.

### ISSUE THREAD

**Issue by saifamp** [BlazingDB/blazingsql#520]
Here is what is going on:

```
from blazingsql import BlazingContext
bc = BlazingContext(allocator="managed")
out = bc.sql("select 1+1;")
```

Gives the following error:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f8ee531b610, pid=3422, tid=0x00007f8f7652f700
#
# JRE version: OpenJDK Runtime Environment (8.0_192-b01) (build 1.8.0_192-b01)
# Java VM: OpenJDK 64-Bit Server VM (25.192-b01 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# C  [_jpype.cpython-37m-x86_64-linux-gnu.so+0x4c610]  JPypeTracer::trace1(char const*)+0x1c0
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /home/ubuntu/hs_err_pid3422.log
#
# If you would like to submit a bug report, please visit:
#   http://www.azulsystems.com/support/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
Aborted (core dumped)
```

This is happening constantly for me. My guess is that I have messed up my installation or something. 

Can anybody guide? Thanks

**Comment by williamBlazing** [comment]
Hello @saifamp BlzingSQL needs to query a table. You need to have to have a table in the FROM statement that you have created using `bc.create_table()`. You would also have an issue if you try to run something like `SELECT 1 + 1, colA FROM tableA` because there in the first column you are just trying to do an operation that does not interact with data from the table. You CAN do `SELECT 2, colA FROM tableA` or you CAN do `SELECT 1+1+colA, colA FROM tableA`.

**Comment by saifamp** [comment]
Hi @williamBlazing . Thanks for the insight. Actually the point of showing `select 1+1` was that it is throwing the same error on the simplest of the queries. I have created a few tables and even when I do a simplest select from those, it does the same thing. 

At this point, I am wondering if I have messed up any of my Java settings, which is causing this JPype error.  Probably I should do a clean re-install (remove everything and reinstall) and see if that changes this. What would be the best way to do so?

**Comment by jsmaupin** [comment]
Is this a duplicate of #512 ?

**Comment by williamBlazing** [comment]
Hello @saifamp that error message will usually occur with almost any crash from the engine, because JPype will also crash when anything else crashes, so it does not necessarily mean the issue is with Java. To help diagnose the issue it would be more helpful if you can provide us with the script you are trying to run, or something more representative. Additionally, if you could tell us about how you installed BlazingSQL (conda, docker, built from source). If conda, please send us the contents of `conda list`. If built from source, please tell us what branch and commit hash.

**Comment by williamBlazing** [comment]
@jsmaupin if is likely not the same situation as #512 since now, if you do have a syntax error, it will still crash, but it will tell you that there was a syntax error before it crashes. We are still working on making it not crash, but be more graceful.

**Comment by saifamp** [comment]
@williamBlazing  Thank you so much for offering help. 

I did install by building from source (For CUDA 10.0 and Python 3.7). I checkout out master and then  ran the build instructions as mentioned on the git page.

The first time I encountered this error was when I tried running my query over the table created from a large parquet file (loaded from cudf as a gdf), without setting allocator as `managed` in the `BlazingContext`. Ever since, any query on any sized table basically throws a similar JPype error. 

Then I thought maybe the build messed up, so I tried from conda (using the conda instructions available on the git page for (For CUDA 10.0 and Python 3.7). This also didnt help.

Here is my small script:

```python
import cudf
from blazingsql import BlazingContext
bc = BlazingContext(allocator="managed")

gdf = cudf.read_parquet("/path/to/parquet.parquet")
bc.create_table('test_table', gdf)
query = "select a, b, count(c) as c from test_table group by 1,2"
bc.sql(query) 
```

Here is `conda list`:

# packages in environment at /home/ubuntu/anaconda3/envs/env1:
#
# Name                    Version                   Build  Channel
_anaconda_depends         2019.03                  py37_0
_libgcc_mutex             0.1                        main
alabaster                 0.7.12                   py37_0
anaconda                  custom                   py37_1
anaconda-client           1.7.2                    py37_0
anaconda-project          0.8.4                      py_0
argh                      0.26.2                   py37_0
arrow-cpp                 0.15.0           py37h5ac5442_0    conda-forge
asn1crypto                1.3.0                    py37_0
astroid                   2.3.3                    py37_0
astropy                   4.0              py37h7b6447c_0
atomicwrites              1.3.0                    py37_1
attrs                     19.3.0                     py_0
autopep8                  1.4.4                      py_0
babel                     2.8.0                      py_0
backcall                  0.1.0                    py37_0
backports                 1.0                        py_2
backports.os              0.1.1           py37hc8dfbb8_1002    conda-forge
backports.shutil_get_terminal_size 1.0.0                    py37_2
beautifulsoup4            4.8.2                    py37_0
bitarray                  1.2.1            py37h7b6447c_0
bkcharts                  0.2                      py37_0
blas                      1.0                         mkl
blazingsql                0.6                       <pip>
blazingsql                0.12.0          cuda10.0_py37_6    blazingsql/label/cuda10.0
bleach                    3.1.0                    py37_0
blosc                     1.17.1               he1b5a44_0    conda-forge
bokeh                     1.4.0                    py37_0
boost-cpp                 1.70.0               ha2d47e9_1    conda-forge
boto                      2.49.0                   py37_0
boto3                     1.12.12                    py_0
botocore                  1.15.12                    py_0
bottleneck                1.3.2            py37heb32a55_0
brotli                    1.0.7             he1b5a44_1001    conda-forge
bsql-engine               0.6                       <pip>
bsql-rapids-thirdparty    0.12.0                        0    blazingsql
bsql-toolchain            0.12.0                        0    blazingsql
bsql-toolchain-aws-cpp    0.12.0                        0    blazingsql
bsql-toolchain-gcp-cpp    0.12.0                        0    blazingsql
bzip2                     1.0.8                h7b6447c_0
c-ares                    1.15.0            h516909a_1001    conda-forge
ca-certificates           2020.1.1                      0
cairo                     1.14.12              h8948797_3
certifi                   2020.4.5.1               py37_0
cffi                      1.14.0           py37h2e261b9_0
chardet                   3.0.4                 py37_1003
click                     7.0                      py37_0
cloudpickle               1.3.0                      py_0
clyent                    1.2.2                    py37_1
cmake                     3.17.0               h28c56e5_0    conda-forge
colorama                  0.4.3                      py_0
contextlib2               0.6.0.post1                py_0
cppzmq                    4.6.0                hc9558a2_0    conda-forge
cryptography              2.8              py37h1ba5d50_0
cudatoolkit               10.0.130                      0    nvidia
cudf                      0.12.0                   py37_0    rapidsai
cudnn                     7.6.0                cuda10.0_0    nvidia
cupy                      7.3.0            py37h658377b_0    conda-forge
curl                      7.68.0               hf8cf82a_0    conda-forge
cycler                    0.10.0                   py37_0
cymem                     1.31.2           py37h6bb024c_0
cyrus-sasl                2.1.27               he38ecfd_0    conda-forge
cython                    0.29.16          py37h3340039_0    conda-forge
cytoolz                   0.9.0.1          py37h14c3975_1
dask                      2.10.1                     py_0
dask-core                 2.10.1                     py_0
dask-cuda                 0.12.0                   py37_0    rapidsai
dask-cudf                 0.12.0                   py37_0    rapidsai
dbus                      1.13.12              h746ee38_0
decorator                 4.4.1                      py_0
defusedxml                0.6.0                      py_0
diff-match-patch          20181111                   py_0
dill                      0.2.9                    py37_0
distributed               2.10.0                     py_0
dlpack                    0.2                  he1b5a44_1    conda-forge
docutils                  0.15.2                   py37_0
double-conversion         3.1.5                he1b5a44_2    conda-forge
entrypoints               0.3                      py37_0
et_xmlfile                1.0.1                    py37_0
expat                     2.2.9                he1b5a44_2    conda-forge
fastavro                  0.23.1           py37h8f50634_0    conda-forge
fastcache                 1.1.0            py37h7b6447c_0
fastrlock                 0.4             py37h3340039_1001    conda-forge
filelock                  3.0.12                     py_0
flake8                    3.7.9                    py37_0
flask                     1.1.1                      py_0
fontconfig                2.13.1            he4413a7_1000    conda-forge
freetype                  2.9.1                h8a8886c_1
fribidi                   1.0.5                h7b6447c_0
fsspec                    0.6.2                      py_0
funcsigs                  1.0.2                     <pip>
future                    0.18.2                   py37_0
get_terminal_size         1.0.0                haa9412d_0
gettext                   0.19.8.1          hc5be6a0_1002    conda-forge
gevent                    1.4.0            py37h7b6447c_0
gflags                    2.2.2             he1b5a44_1002    conda-forge
glib                      2.63.1               h5a9c865_0
glob2                     0.7                        py_0
glog                      0.4.0                he1b5a44_1    conda-forge
gmock                     1.10.0                        2    conda-forge
gmp                       6.1.2                h6c8ec71_1
gmpy2                     2.0.8            py37h10f8cd9_2
graphite2                 1.3.13               h23475e2_0
greenlet                  0.4.15           py37h7b6447c_0
grpc-cpp                  1.23.0               h18db393_0    conda-forge
gst-plugins-base          1.14.0               hbbd80ab_1
gstreamer                 1.14.0               hb453b48_1
gtest                     1.10.0               hc9558a2_2    conda-forge
h5py                      2.10.0           py37h7918eee_0
harfbuzz                  1.8.8                hffaf4a1_0
hdf5                      1.10.4               hb1b8bf9_0
heapdict                  1.0.1                      py_0
html5lib                  1.0.1                    py37_0
hypothesis                5.5.4                      py_0
icu                       58.2                 h9c2bf20_1
idna                      2.8                      py37_0
imageio                   2.6.1                    py37_0
imagesize                 1.2.0                      py_0
importlib_metadata        1.5.0                    py37_0
intel-openmp              2020.0                      166
intervaltree              3.0.2                      py_0
ipykernel                 5.1.4            py37h39e3cac_0
ipython                   7.12.0           py37h5ca1d4c_0
ipython_genutils          0.2.0                    py37_0
ipywidgets                7.5.1                      py_0
isort                     4.3.21                   py37_0
itsdangerous              1.1.0                    py37_0
jbig                      2.1                  hdba287a_0
jdcal                     1.4.1                      py_0
jedi                      0.14.1                   py37_0
jeepney                   0.4.2                      py_0
jinja2                    2.11.1                     py_0
jmespath                  0.9.4                      py_0
joblib                    0.14.1                     py_0
jpeg                      9b                   h024ee3a_2
jpype1                    0.7.2            py37hc9558a2_0    conda-forge
json5                     0.9.1                      py_0
jsonschema                3.2.0                    py37_0
jupyter                   1.0.0                    py37_7
jupyter_client            5.3.4                    py37_0
jupyter_console           6.1.0                      py_0
jupyter_core              4.6.1                    py37_0
jupyterlab                1.2.6              pyhf63ae98_0
jupyterlab_server         1.0.6                      py_0
keyring                   21.1.0                   py37_0
kiwisolver                1.1.0            py37he6710b0_0
krb5                      1.16.4               h2fd8d38_0    conda-forge
lazy-object-proxy         1.4.3            py37h7b6447c_0
ld_impl_linux-64          2.33.1               h53a641e_7
libarchive                3.3.3             hb44662c_1005    conda-forge
libcudf                   0.12.0               cuda10.0_0    rapidsai
libcurl                   7.68.0               hda55be3_0    conda-forge
libedit                   3.1.20181209         hc058e9b_0
libevent                  2.1.10               h72c5cf5_0    conda-forge
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 9.1.0                hdf63c60_0
libgcrypt                 1.8.4             hf484d3e_1000    conda-forge
libgfortran-ng            7.3.0                hdf63c60_0
libgpg-error              1.36                 he1b5a44_0    conda-forge
libgsasl                  1.8.0             h19a2143_1004    conda-forge
libhdfs3                  2.3               h311b756_1006    conda-forge
liblief                   0.9.0                h7725739_2
libntlm                   1.4               h516909a_1002    conda-forge
libnvstrings              0.12.0               cuda10.0_0    rapidsai
libpng                    1.6.37               hbc83047_0
libprotobuf               3.8.0                h8b12597_0    conda-forge
librmm                    0.12.0               cuda10.0_0    rapidsai
libsodium                 1.0.17               h516909a_0    conda-forge
libspatialindex           1.9.3                he6710b0_0
libssh2                   1.8.2                h1ba5d50_0
libstdcxx-ng              9.1.0                hdf63c60_0
libtiff                   4.0.9                he6b73bb_1    conda-forge
libtool                   2.4.6                h7b6447c_5
libuuid                   2.32.1            h14c3975_1000    conda-forge
libuv                     1.34.0               h516909a_0    conda-forge
libxcb                    1.13                 h1bed415_1
libxml2                   2.9.9                hea5a465_1
libxslt                   1.1.33               h7d1a2b0_0
llvmlite                  0.31.0           py37hd408876_0
locket                    0.2.0                    py37_1
lxml                      4.5.0            py37hefd8a0e_0
lz4-c                     1.8.3             he1b5a44_1001    conda-forge
lzo                       2.10                 h49e0be7_2
markupsafe                1.1.1            py37h7b6447c_0
matplotlib                3.1.3                    py37_0
matplotlib-base           3.1.3            py37hef1b27d_0
maven                     3.6.0                         0    conda-forge
mccabe                    0.6.1                    py37_1
mistune                   0.8.4            py37h7b6447c_0
mkl                       2020.0                      166
mkl-service               2.3.0            py37he904b0f_0
mkl_fft                   1.0.15           py37ha843d7b_0
mkl_random                1.1.0            py37hd6b4f25_0
mock                      4.0.1                      py_0
modin                     0.7.2                     <pip>
more-itertools            8.2.0                      py_0
mpc                       1.1.0                h10f8cd9_1
mpfr                      4.0.1                hdf1c602_3
mpmath                    1.1.0                    py37_0
msgpack-numpy             0.4.4.3                    py_0
msgpack-python            0.5.6            py37h6bb024c_1
multipledispatch          0.6.0                    py37_0
murmurhash                0.28.0           py37hf484d3e_0
nbconvert                 5.6.1                    py37_0
nbformat                  5.0.4                      py_0
nccl                      2.4.6.1              cuda10.0_0    nvidia
ncurses                   6.1               hf484d3e_1002    conda-forge
netifaces                 0.10.9          py37h8f50634_1002    conda-forge
networkx                  2.4                        py_0
nltk                      3.4.5                    py37_0
nose                      1.3.7                    py37_2
notebook                  6.0.3                    py37_0
numba                     0.48.0           py37h0573a6f_0
numexpr                   2.7.1            py37h423224d_0
numpy                     1.18.1           py37h4f9e942_0
numpy-base                1.18.1           py37hde5b4d6_1
numpydoc                  0.9.2                      py_0
nvstrings                 0.12.0                   py37_0    rapidsai
olefile                   0.46                     py37_0
openjdk                   8.0.192           h516909a_1004    conda-forge
openpyxl                  3.0.3                      py_0
openssl                   1.1.1f               h7b6447c_0
packaging                 20.1                       py_0
pandas                    0.25.3           py37hb3f55d8_0    conda-forge
pandas                    1.0.1                     <pip>
pandoc                    2.2.3.2                       0
pandocfilters             1.4.2                    py37_1
pango                     1.40.14              he752989_2    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.5.2                      py_0
partd                     1.1.0                      py_0
patchelf                  0.10                 he6710b0_0
path                      13.1.0                   py37_0
path.py                   12.4.0                        0
pathlib2                  2.3.5                    py37_0
pathtools                 0.1.2                      py_1
patsy                     0.5.1                    py37_0
pcre                      8.43                 he6710b0_0
pep8                      1.7.1                    py37_0
pexpect                   4.8.0                    py37_0
pickleshare               0.7.5                    py37_0
pillow                    5.4.1            py37h34e0f95_0
pip                       20.0.2                   py37_1
pixman                    0.38.0               h7b6447c_0
pkginfo                   1.5.0.1                  py37_0
plac                      0.9.6                    py37_0
pluggy                    0.13.1                   py37_0
ply                       3.11                     py37_0
preshed                   1.0.1            py37he6710b0_0
prometheus_client         0.7.1                      py_0
prompt_toolkit            3.0.3                      py_0
protobuf                  3.11.3                    <pip>
psutil                    5.6.7            py37h7b6447c_0
ptyprocess                0.6.0                    py37_0
py                        1.8.1                      py_0
py-lief                   0.9.0            py37h7725739_2
pyarrow                   0.15.0           py37h8b68381_1    conda-forge
pycodestyle               2.5.0                    py37_0
pycosat                   0.6.3            py37h7b6447c_0
pycparser                 2.19                     py37_0
pycrypto                  2.6.1            py37h14c3975_9
pycurl                    7.43.0.5         py37h1ba5d50_0
pydocstyle                4.0.1                      py_0
pyflakes                  2.1.1                    py37_0
pygments                  2.5.2                      py_0
pyhive                    0.6.1                    py37_0
pylint                    2.4.4                    py37_0
pynvml                    8.0.4                      py_0    conda-forge
pyodbc                    4.0.30           py37he6710b0_0
pyopenssl                 19.1.0                   py37_0
pyparsing                 2.4.6                      py_0
pyqt                      5.9.2            py37h05f1152_2
pyrsistent                0.15.7           py37h7b6447c_0
pysocks                   1.7.1                    py37_0
pytables                  3.6.1            py37h71ec239_0
pytest                    5.4.1            py37hc8dfbb8_0    conda-forge
pytest-arraydiff          0.3              py37h39e3cac_0
pytest-astropy            0.8.0                      py_0
pytest-astropy-header     0.1.2                      py_0
pytest-doctestplus        0.5.0                      py_0
pytest-openfiles          0.4.0                      py_0
pytest-remotedata         0.3.2                    py37_0
python                    3.7.6           h8356626_5_cpython    conda-forge
python-dateutil           2.8.1                      py_0
python-jsonrpc-server     0.3.4                      py_0
python-language-server    0.31.7                   py37_0
python-libarchive-c       2.8                     py37_13
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2019.3                     py_0
pywavelets                1.1.1            py37h7b6447c_0
pyxdg                     0.26                       py_0
pyyaml                    5.3              py37h7b6447c_0
pyzmq                     18.1.1           py37he6710b0_0
qdarkstyle                2.8                        py_0
qt                        5.9.7                h5867ecd_1
qtawesome                 0.6.1                      py_0
qtconsole                 4.6.0                      py_1
qtpy                      1.9.0                      py_0
rapidjson                 1.1.0             he1b5a44_1002    conda-forge
ray                       0.8.0                     <pip>
re2                       2020.04.01           he1b5a44_0    conda-forge
readline                  8.0                  hf8c457e_0    conda-forge
redis                     3.4.1                     <pip>
regex                     2018.07.11       py37h14c3975_0
requests                  2.22.0                   py37_1
rhash                     1.3.6             h14c3975_1001    conda-forge
ripgrep                   11.0.2               he32d670_0
rmm                       0.12.0                   py37_0    rapidsai
rope                      0.16.0                     py_0
rtree                     0.9.3                    py37_0
ruamel_yaml               0.15.87          py37h7b6447c_0
s3transfer                0.3.3                    py37_0
sasl                      0.2.1           py37h3340039_1002    conda-forge
scikit-image              0.16.2           py37h0573a6f_0
scikit-learn              0.22.1           py37hd81dba3_0
scipy                     1.4.1            py37h0b6359f_0
seaborn                   0.10.0                     py_0
secretstorage             3.1.2                    py37_0
send2trash                1.5.0                    py37_0
setuptools                45.2.0                   py37_0
simplegeneric             0.8.1                    py37_2
singledispatch            3.4.0.3                  py37_0
sip                       4.19.8           py37hf484d3e_0
six                       1.14.0                   py37_0
snappy                    1.1.7                hbae5bb6_3
snowballstemmer           2.0.0                      py_0
sortedcollections         1.1.2                    py37_0
sortedcontainers          2.1.0                    py37_0
soupsieve                 1.9.5                    py37_0
spacy                     2.0.12           py37h962f231_0
sphinx                    2.4.0                      py_0
sphinxcontrib             1.0                      py37_1
sphinxcontrib-applehelp   1.0.1                      py_0
sphinxcontrib-devhelp     1.0.1                      py_0
sphinxcontrib-htmlhelp    1.0.2                      py_0
sphinxcontrib-jsmath      1.0.1                      py_0
sphinxcontrib-qthelp      1.0.2                      py_0
sphinxcontrib-serializinghtml 1.1.3                      py_0
sphinxcontrib-websupport  1.2.0                      py_0
spyder                    4.0.1                    py37_0
spyder-kernels            1.8.1                    py37_0
sqlalchemy                1.3.13           py37h7b6447c_0
sqlite                    3.31.1               h7b6447c_0
statsmodels               0.11.0           py37h7b6447c_0
sympy                     1.5.1                    py37_0
tbb                       2020.0               hfd86e86_0
tblib                     1.6.0                      py_0
termcolor                 1.1.0                    py37_1
terminado                 0.8.3                    py37_0
testpath                  0.4.4                      py_0
thinc                     6.10.3           py37h962f231_0
thrift                    0.13.0           py37h3340039_1    conda-forge
thrift-cpp                0.12.0            hf3afdfd_1004    conda-forge
thrift_sasl               0.4.2            py37h8f50634_0    conda-forge
tk                        8.6.10               hed695b0_0    conda-forge
toolz                     0.10.0                     py_0
tornado                   6.0.3            py37h7b6447c_3
tqdm                      4.42.1                     py_0
traitlets                 4.3.3                    py37_0
ujson                     1.35             py37h14c3975_0
unicodecsv                0.14.1                   py37_0
unixodbc                  2.3.7                h14c3975_0
uriparser                 0.9.3                he1b5a44_1    conda-forge
urllib3                   1.25.8                   py37_0
watchdog                  0.10.2                   py37_0
wcwidth                   0.1.8                      py_0
webencodings              0.5.1                    py37_1
werkzeug                  1.0.0                      py_0
wheel                     0.34.2                   py37_0
widgetsnbextension        3.5.1                    py37_0
wrapt                     1.10.11          py37h14c3975_2
wurlitzer                 2.0.0                    py37_0
xlrd                      1.2.0                    py37_0
xlsxwriter                1.2.7                      py_0
xlwt                      1.3.0                    py37_0
xz                        5.2.4                h14c3975_4
yaml                      0.1.7                had09818_2
yapf                      0.28.0                     py_0
zeromq                    4.3.2                he1b5a44_2    conda-forge
zict                      1.0.0                      py_0
zipp                      2.2.0                      py_0
zlib                      1.2.11               h7b6447c_3
zstd                      1.4.0                h3b9ef0a_0    conda-forge

**Comment by williamBlazing** [comment]
Hello @saifamp Sorry for the late reply, but it seems like you are using the old version of our engine. I would suggest creating a new environment and installing the latest  version which was released not that long ago. https://github.com/BlazingDB/blazingsql/#stable-version

**Comment by saifamp** [comment]
Thank you @williamBlazing . :-) I will try and post here.

**Comment by williamBlazing** [comment]
Closing issue for now @saifamp if you still are having an issue, please feel free to re-open or create a new issue
